libs/queue/README.md

```diff
@@ -0,0 +1,64 @@
+# Overview
+
+The Queue library provides an implementation of a queue data structure. A Queue is a linear structure which follows the First-In-First-Out (FIFO) principle. This means that the elements added first are the ones that get removed first.
+
+The `Queue<T>` is a structure with a generic type parameter T. The elements in the queue are stored as a `Vec<T>`. The data structure is created empty and is growable.
+
+For more information please see the [specification](./SPECIFICATION.md).
+
+> **Note** The Queue<T> type in this library is different from built-in arrays and vectors due to its unique characteristics, especially its operation in FIFO manner.
```

Comment:
I would also remove this note as it does not inform the user of anything new / add value. It has been stated in the first sentence that it's a structure that follows FIFO and here it re-iterates that and points out that it's not a _different_ data structure - which is not something that someone would automatically think e.g. when thinking of an unsigned integer and a signed integer you may think about the commonality of them being integers; nevertheless, they are different and not interchangeable

Reply:
I will be making some adjustments to make sure the SPECIFICATION.md has new information that can't be found on the README.md. Thank you for pointing that one out. As for the suggestion about stating that Queues are different from the built-in arrays and vectors, are you recommending that I remove that portion of the note? I would think that it's worth mentioning that it is different than the built-in data structures but would like to know your thoughts.

Reply:
I would remove it.
I don't see why someone would assume that two different things are the same - even if they are ignorant.

If someone wants to use the library then it's likely that they already have an idea of what it is; moreover, if they do not know then showing them how to use the library should be sufficient. If you do not show them how to use a queue as an array (because it may not work that way) then they should not assume that they can use it that way.

As another example, if you want to purchase a car do you expect the salesperson to say "by the way, it does not fly because it's not a plane"?
There's no reason to state that information.

Reply:
I see your point with that. I'll take it out.